src/test/java/com/vinyl/RemoveProductFromStoreTest.java

```diff
@@ -0,0 +1,95 @@
+package com.vinyl;
+
+import com.vinyl.helper.TokenHeaderHelper;
+import com.vinyl.model.Product;
+import com.vinyl.repository.ProductRepository;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.*;
+
+import java.time.LocalDate;
+
+public class RemoveProductFromStoreTest extends ManagerBaseIntegration {
+
+    @Autowired
+    protected TokenHeaderHelper tokenHeaderHelper;
+
+    @Autowired
+    protected ProductRepository productRepository;
+
+    protected Product product;
+
+    @Override
+    public void setUp() {
+        super.setUp();
+        product = defaultEntitiesHelper.createProduct();
+    }
+
+    @Test
+    public void testWhenOK() {
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.NO_CONTENT);
+    }
+
+    @Test
+    public void testWhenProductIdIsInvalid() {
+        Integer invalidId = 00;
+        product.setId(invalidId);
+        productRepository.save(product);
+
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.NOT_FOUND);
```

Comment:
response should be BAD_REQUEST, also assert that that product not removed from db

Reply:
done